### ISSUE TITLE
Old name of renamed file shows up in Got to References

### ISSUE THREAD

**Issue by PEZ** [snoe/clojure-lsp#115]
In vscode:

1. I copied a file, `foo.clj`
1. which gave me `foo copy.clj`
1. I then renamed it to `bar.clj`
1. Then in `foo.clj` I asked for **Go to references**

**Expected:**
* The list includes `bar.clj`
* The list _does not include_ `foo copy.clj`, since it does no longer exist

**Result:**
* The list includes `bar.clj`
* The list also _includes_ `foo copy.clj`, even though it does no longer exist under that name.

**Comment by snoe** [comment]
Fixed in c8f7e3068293b23c305450eb881dbc925ae97a17

I decided to let the client inform clojure-lsp when a file has been deleted through the `workspace/didChangeWatchedFiles` protocol. We now send a `client/registerCapability` for the method after initialization. Clients who support the capability should be able to set the `workspace/didChangeWatchedFiles` messages back to the server and we will remove any deleted files from the clojure-lsp db.
